src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1667,6 +1667,36 @@ class Simplex_tree {
     return sh; // None of its faces has the same filtration.
   }
 
+ public:
+  /** \brief This function resets filtration value until a given dimension.
+   * @param[in] filt_value The new filtration value.
+   * @param[in] max_dim The maximal dimension.
+   */
+  void reset_filtration(Filtration_value filt_value, int max_dim) {
```

Comment:
I think a minimal dimension may be more useful than a maximum. For instance, I set some values values on the vertices, reset all other simplices, and call make_filtration_non_decreasing. We can still keep max_dim, but it should have a default value (INT_MAX?).

Reply:
Hm ok... Maybe more tricky, but doable...
And why not resetting all simplices and then set some values on the vertices ? There is a need ?
Ok for max_dim default value.